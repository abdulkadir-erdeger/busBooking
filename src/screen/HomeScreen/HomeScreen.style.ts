import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: { flex: 1 },
  slideBar: { margin: 2 },
  slidePicture: { width: "100%", height: 250 },
  slideText: {
    width: "100%",
    opacity: 0.5,
    backgroundColor: "black",
    position: "absolute",
    bottom: 0,
    fontSize: 20,
    padding: 8,
    fontWeight: "bold",
    color: "white",
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
  },
  travelContainer: { flex: 1, padding: 5 },
  radioButtonContainer: { flexDirection: "row", alignItems: "center" },
  cityInput: {
    marginHorizontal: 5,
    marginVertical: 5,
    paddingStart: 15,
    borderRadius: 10,
    color: "black",
    borderWidth: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  Input: {
    flexDirection: "row",
    alignItems: "center",
    marginHorizontal: 5,
    marginVertical: 5,
    paddingStart: 15,
    borderRadius: 10,
    color: "black",
    borderWidth: 1,
    padding: 16,
  },
  button: {
    marginHorizontal: 10,
    backgroundColor: "#f9a825",
    marginVertical: 10,
    padding: 10,
    alignItems: "center",
    borderRadius: 15,
  },
  buttonText: {
    color: "white",
    fontWeight: "bold",
    fontSize: 18,
  },
});
