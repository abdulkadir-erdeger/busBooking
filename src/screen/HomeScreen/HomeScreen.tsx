import React, { useState } from "react";
import { View, Text, StatusBar, Image, Pressable, LogBox } from "react-native";
import { RadioButton } from "react-native-paper";
import { Picker } from "@react-native-picker/picker";
import DateTimePicker, {
  DateTimePickerEvent,
} from "@react-native-community/datetimepicker";
import { MaterialCommunityIcons, Ionicons } from "@expo/vector-icons";
import { useNavigation } from "@react-navigation/native";
import styles from "./HomeScreen.style";
import { RootStackParamList } from "../../types";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";

LogBox.ignoreLogs(["new NativeEventEmitter"]);
LogBox.ignoreAllLogs();

const HomeScreen = () => {
  const navigation =
    useNavigation<NativeStackNavigationProp<RootStackParamList>>();
  const [checked, setChecked] = useState<string>("tekYon");
  const [departureCity, setDepartureCity] = useState<string>("istanbul");
  const [destinationCity, setDestinationCity] = useState<string>("ankara");
  const [maturityStatus, setMaturityStatus] = useState<string>("yetiskin");
  const [date, setDate] = useState(new Date());
  const [show, setShow] = useState<Boolean>(false);

  const onChange = (event: DateTimePickerEvent, selectedDate: any) => {
    const currentDate = selectedDate;
    setShow(false);
    setDate(currentDate);
  };

  const months = [
    "Ocak",
    "Şubat",
    "Mart",
    "Nisan",
    "Mayıs",
    "Haziran",
    "Temmuz",
    "Ağustos",
    "Eylül",
    "Ekim",
    "Kasım",
    "Aralık",
  ];
  const days = [
    "Pazar",
    "Pazartesi",
    "Salı",
    "Çarşamba",
    "Perşembe",
    "Cuma",
    "Cumartesi",
  ];

  function info() {
    const obj = {
      biletTürü: checked,
      kalkıs: departureCity,
      varis: destinationCity,
      haftaninGunu: days[date.getDay()],
      gun: date.getDate(),
      ay: months[date.getMonth()],
      yasDurum: maturityStatus,
    };
    return obj;
  }

  const navigateToDetail = (): void => {
    navigation.navigate("Expedition", info());
  };

  return (
    <View style={styles.container}>
      <StatusBar barStyle={"light-content"} />
      <View style={styles.slideBar}>
        <Image
          style={styles.slidePicture}
          source={require("../../../assets/bus.png")}
        />
        <Text style={styles.slideText}>
          En Uygun Fiyatlı Otobüs Biletleri Yolcudur Abbas da
        </Text>
      </View>
      <View style={styles.travelContainer}>
        <View style={styles.radioButtonContainer}>
          <RadioButton
            value="tekYon"
            status={checked === "tekYon" ? "checked" : "unchecked"}
            onPress={() => setChecked("tekYon")}
          />
          <Text>Tek Yön</Text>
          <RadioButton
            value="gidisDonus"
            status={checked === "gidisDonus" ? "checked" : "unchecked"}
            onPress={() => setChecked("gidisDonus")}
          />
          <Text>Gidiş Dönüş</Text>
        </View>

        <View style={styles.cityInput}>
          <MaterialCommunityIcons
            name="map-marker-outline"
            size={24}
            color="black"
          />
          <Picker
            style={{ flex: 1 }}
            selectedValue={departureCity}
            onValueChange={(itemValue, itemIndex) =>
              setDepartureCity(itemValue)
            }
          >
            <Picker.Item label="İstanbul" value="istanbul" />
            <Picker.Item label="Ankara" value="ankara" />
          </Picker>
        </View>

        <View style={styles.cityInput}>
          <MaterialCommunityIcons
            name="map-marker-outline"
            size={24}
            color="black"
          />
          <Picker
            style={{ flex: 1 }}
            selectedValue={destinationCity}
            onValueChange={(itemValue, itemIndex) =>
              setDestinationCity(itemValue)
            }
          >
            <Picker.Item label="Ankara" value="ankara" />
            <Picker.Item label="İstanbul" value="istanbul" />
          </Picker>
        </View>

        <Pressable onPress={() => setShow(true)}>
          <View style={styles.Input}>
            <Ionicons
              style={{ marginEnd: 10 }}
              name="calendar-outline"
              size={24}
              color="black"
            />
            <Text>{date.toLocaleDateString()}</Text>
          </View>
        </Pressable>

        <View style={styles.cityInput}>
          <Picker
            style={{ flex: 1 }}
            selectedValue={maturityStatus}
            onValueChange={(itemValue, itemIndex) =>
              setMaturityStatus(itemValue)
            }
          >
            <Picker.Item label="Yetişkin" value="yetiskin" />
            <Picker.Item label="Çocuk" value="cocuk" />
          </Picker>
        </View>
      </View>

      {show && (
        <DateTimePicker
          testID="dateTimePicker"
          value={date}
          mode="date"
          is24Hour={true}
          onChange={onChange}
        />
      )}

      <Pressable
        testID="buttonNavigate"
        onPress={() => navigateToDetail()}
        style={styles.button}
      >
        <Text style={styles.buttonText}>Seferleri Göster</Text>
      </Pressable>
    </View>
  );
};

export default HomeScreen;
