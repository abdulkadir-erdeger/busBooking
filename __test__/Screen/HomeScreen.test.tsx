import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import HomeScreen from "../../src/screen/HomeScreen";

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

test("render correctly", () => {
  render(<HomeScreen />);
  expect(screen.toJSON()).toMatchSnapshot();
});

const route = {
  ay: "Kasım",
  biletTürü: "tekYon",
  gun: 24,
  haftaninGunu: "Perşembe",
  kalkıs: "istanbul",
  varis: "ankara",
  yasDurum: "yetiskin",
};

test("uses useNavigation when pressed", async () => {
  const { getAllByTestId } = render(<HomeScreen />);
  const button = getAllByTestId("buttonNavigate");
  await fireEvent.press(button[0]);
  expect(mockedNavigate).toHaveBeenCalledWith("Expedition", route);
});
