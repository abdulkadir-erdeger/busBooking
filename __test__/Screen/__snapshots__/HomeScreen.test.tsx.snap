// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "margin": 2,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 250,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        {
          "backgroundColor": "black",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "bottom": 0,
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "opacity": 0.5,
          "padding": 8,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      En Uygun Fiyatlı Otobüs Biletleri Yolcudur Abbas da
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "checked": true,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
      <Text>
        Tek Yön
      </Text>
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "checked": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 0,
            }
          }
        >
          <Text />
        </View>
      </View>
      <Text>
        Gidiş Dönüş
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "flexDirection": "row",
          "marginHorizontal": 5,
          "marginVertical": 5,
          "paddingStart": 15,
        }
      }
    >
      <Text />
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNCPicker
          items={
            [
              {
                "label": "İstanbul",
                "testID": undefined,
                "textColor": undefined,
                "value": "istanbul",
              },
              {
                "label": "Ankara",
                "testID": undefined,
                "textColor": undefined,
                "value": "ankara",
              },
            ]
          }
          numberOfLines={1}
          onChange={[Function]}
          selectedIndex={0}
          style={
            [
              {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "flexDirection": "row",
          "marginHorizontal": 5,
          "marginVertical": 5,
          "paddingStart": 15,
        }
      }
    >
      <Text />
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNCPicker
          items={
            [
              {
                "label": "Ankara",
                "testID": undefined,
                "textColor": undefined,
                "value": "ankara",
              },
              {
                "label": "İstanbul",
                "testID": undefined,
                "textColor": undefined,
                "value": "istanbul",
              },
            ]
          }
          numberOfLines={1}
          onChange={[Function]}
          selectedIndex={0}
          style={
            [
              {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "black",
            "flexDirection": "row",
            "marginHorizontal": 5,
            "marginVertical": 5,
            "padding": 16,
            "paddingStart": 15,
          }
        }
      >
        <Text />
        <Text>
          24.11.2022
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "flexDirection": "row",
          "marginHorizontal": 5,
          "marginVertical": 5,
          "paddingStart": 15,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNCPicker
          items={
            [
              {
                "label": "Yetişkin",
                "testID": undefined,
                "textColor": undefined,
                "value": "yetiskin",
              },
              {
                "label": "Çocuk",
                "testID": undefined,
                "textColor": undefined,
                "value": "cocuk",
              },
            ]
          }
          numberOfLines={1}
          onChange={[Function]}
          selectedIndex={0}
          style={
            [
              {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f9a825",
        "borderRadius": 15,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    testID="buttonNavigate"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Seferleri Göster
    </Text>
  </View>
</View>
`;
